shared/actions/profile.js

```diff
@@ -63,20 +72,19 @@ function waitingForResponse (waiting: boolean): Waiting {
   }
 }
 
-function selectPlatform (platform: PlatformsExpanded): UpdatePlatform {
+function updatePlatform (platform: PlatformsExpandedType): UpdatePlatform {
   return {
     type: Constants.updatePlatform,
     payload: {platform},
   }
 }
 
-let submitUsernameResponse: ?Object = null
-
 function submitUsername (): AsyncAction {
   return (dispatch, getState) => {
-    if (submitUsernameResponse) {
-      submitUsernameResponse.result(getState().profile.username)
-      submitUsernameResponse = null
+    if (promptUsernameResponse) {
+      dispatch(updateError(null))
+      promptUsernameResponse.result(getState().profile.username)
```

Comment:
i doubt it. this is for all services so its hard to validate what would be good or not. we could check if username.length > 0 and thats about it right?